src/platform_impl/windows/window.rs

```diff
@@ -405,49 +423,44 @@ impl Window {
                 {
                     let monitor = video_mode.monitor();
 
-                    let mut display_name = OsStr::new(&monitor.inner.native_identifier())
-                        .encode_wide()
-                        .collect::<Vec<_>>();
-                    // `encode_wide` does not add a null-terminator but
-                    // `ChangeDisplaySettingsExW` requires a null-terminated
-                    // string, so add it
-                    display_name.push(0);
-
-                    let mut native_video_mode = video_mode.video_mode.native_video_mode;
+                    let native_video_mode = video_mode.video_mode.native_video_mode;
 
                     let res = unsafe {
-                        winuser::ChangeDisplaySettingsExW(
-                            display_name.as_ptr(),
-                            &mut native_video_mode,
-                            std::ptr::null_mut(),
-                            winuser::CDS_FULLSCREEN,
+                        ChangeDisplaySettingsExW(
+                            monitor::get_monitor_info(monitor.inner.hmonitor())
+                                .unwrap()
+                                .szDevice
+                                .as_mut_ptr(),
```

Comment:
Taking `.as_mut_ptr()` here *feels* wrong, even tough it's probably fine.

Reply:
I could do `as_ptr() as PWSTR`?

Reply:
It's more that it's taking a pointer to a temporary variable.

Reply:
Extracted the `monitor_info` into a variable. The lifetime should now be the same as before.